packages/core/src/odata/common/uri-conversion/get-filter.ts

```diff
@@ -130,6 +131,25 @@ export function createGetFilter(uriConverter: UriConverter) {
         uriConverter.convertToUriFormat(filter.value, filter.edmType!)
       ].join(' ');
     }
+
+    if (isFilterLambdaExpression(filter)) {
+      const alias = 'a';
+      filter.validate();
+      if (parentFieldNames.length !== 1) {
+        throw new Error(
+          `The number of the parent fields: ${parentFieldNames} must be 1, when building a filter lambda expression.`
+        );
+      }
+      const filterExp = filter.filters
+        .map(subFilter =>
+          getODataFilterExpression(subFilter, [], targetEntityConstructor)
+        )
+        .filter(f => !!f)
+        .join(' and ');
```

Comment:
This is still duplicate code. Instead wrap a filterlist around the expressions in the constructor of the lambda expression (see below).

```suggestion
      const filterExp = getODataFilterExpression(filter.filter, [], targetEntityConstructor);
```